model/client4.go

```diff
@@ -2235,6 +2235,16 @@ func (c *Client4) SearchChannels(teamId string, search *ChannelSearch) ([]*Chann
 	return ChannelSliceFromJson(r.Body), BuildResponse(r)
 }
 
+// SearchArchivedChannels returns the archived channels on a team matching the provided search term.
+func (c *Client4) SearchArchivedChannels(teamId string, search *ChannelSearch, userId string) ([]*Channel, *Response) {
```

Comment:
`userId` isn't used here, so I think we can remove it. We don't need the methods of `Client4` to match one-to-one to the API